### PR TITLE
Update Quantization docs and minor doc fix.

### PR DESCRIPTION
**Pull request by Vaibhavs10**
Updates mostly from chats w/ @ErikKaum & @danieldk - feel free to suggest edits 🙏

### CODE REVIEW

**Review by ErikKaum — DISMISSED** [review]
Thank you @Vaibhavs10 🚀 
To me these are a clear improvement, nothing to add at this stage 👍

File: README.md [hunk 1]
```diff
@@ -48,6 +48,8 @@ Text Generation Inference (TGI) is a toolkit for deploying and serving Large Lan
   - [GPT-Q](https://arxiv.org/abs/2210.17323)
   - [EETQ](https://github.com/NetEase-FuXi/EETQ)
   - [AWQ](https://github.com/casper-hansen/AutoAWQ)
+  - [Marlin](https://github.com/IST-DASLab/marlin)
+  - [fp8]()
```
Comment (Vaibhavs10) [inline README.md:52]: Any idea what should I link to here for `fp8`?

File: docs/source/basic_tutorials/launcher.md [hunk 2]
```diff
@@ -55,16 +55,16 @@ Options:
 ## QUANTIZE
 ```shell
       --quantize <QUANTIZE>
-          Whether you want the model to be quantized
+          Whether you want the model to be quantized/ load a pre-quantized model.
```
Comment (pcuenca) [inline docs/source/basic_tutorials/launcher.md:58]:
```suggestion
          Whether you want the model to be quantized / load a pre-quantized model.
```

File: docs/source/basic_tutorials/preparing_model.md [hunk 3]
```diff
@@ -4,7 +4,7 @@ Text Generation Inference improves the model in several aspects.
...
+TGI supports [bits-and-bytes](https://github.com/TimDettmers/bitsandbytes#bitsandbytes), [GPT-Q](https://arxiv.org/abs/2210.17323), [AWQ](https://arxiv.org/abs/2306.00978), [Marlin](https://github.com/IST-DASLab/marlin), [EETQ](https://github.com/NetEase-FuXi/EETQ), [EXL2](https://github.com/turboderp/exllamav2), and [fp8](https://developer.nvidia.com/blog/nvidia-arm-and-intel-publish-fp8-specification-for-standardization-as-an-interchange-format-for-ai/) quantization. To speed up inference with quantization, simply set `quantize` flag to `bitsandbytes`, `gptq`, `awq`, `marlin`, `exl2`, `eetq` or `fp8` depending on the quantization technique you wish to use. When using GPT-Q quantization, you need to point to one of the models [here](https://huggingface.co/models?search=gptq) when using AWQ quantization, you need to point to one of the models [here](https://huggingface.co/models?search=awq). To get more information about quantization, please refer to [quantization guide](./../conceptual/quantization)
```
Comment (pcuenca) [inline docs/source/basic_tutorials/preparing_model.md:7]:
```suggestion
TGI supports [bits-and-bytes](https://github.com/TimDettmers/bitsandbytes#bitsandbytes), [GPT-Q](https://arxiv.org/abs/2210.17323), [AWQ](https://arxiv.org/abs/2306.00978), [Marlin](https://github.com/IST-DASLab/marlin), [EETQ](https://github.com/NetEase-FuXi/EETQ), [EXL2](https://github.com/turboderp/exllamav2), and [fp8](https://developer.nvidia.com/blog/nvidia-arm-and-intel-publish-fp8-specification-for-standardization-as-an-interchange-format-for-ai/) quantization. To speed up inference with quantization, simply set `quantize` flag to `bitsandbytes`, `gptq`, `awq`, `marlin`, `exl2`, `eetq` or `fp8` depending on the quantization technique you wish to use. When using GPT-Q quantization, you need to point to one of the models [here](https://huggingface.co/models?search=gptq). Similarly, when using AWQ quantization, you need to point to one of [these models](https://huggingface.co/models?search=awq). To get more information about quantization, please refer to [quantization guide](./../conceptual/quantization)
```

Comment (pcuenca) [reply docs/source/basic_tutorials/preparing_model.md:7]: Same for Marlin and EXL2, no?

File: docs/source/conceptual/quantization.md [hunk 4]
```diff
@@ -1,6 +1,40 @@
 # Quantization
 
-TGI offers GPTQ and bits-and-bytes quantization to quantize large language models.
+TGI offers many quantization schemes to run LLMs effectively and fast based on your use-case. TGI supports GPTQ, AWQ, bits-and-bytes, EETQ, Marlin, EXL2 and fp8 quantization.
+
+To leverage GPTQ, AWQ, Marlin and EXL2 quants, you must provide pre-quantized weights. Where as for bits-and-bytes, EETQ and fp8 the weights are quantized on the fly by TGI.
```
Comment (pcuenca) [inline docs/source/conceptual/quantization.md:5]:
```suggestion
To leverage GPTQ, AWQ, Marlin and EXL2 quants, you must provide pre-quantized weights. Whereas for bits-and-bytes, EETQ and fp8, weights are quantized by TGI on the fly.
```

File: docs/source/conceptual/quantization.md [hunk 5]
```diff
@@ -1,6 +1,40 @@
...
+
+We recommend using the official quantization scripts for creating your quants:
+1. [AWQ](https://github.com/casper-hansen/AutoAWQ/blob/main/examples/quantize.py)
+2. [GPTQ/ Marlin](https://github.com/AutoGPTQ/AutoGPTQ/blob/main/examples/quantization/basic_usage.py)
+3. [EXL2](https://github.com/turboderp/exllamav2/blob/master/doc/convert.md)
+
+For on-the-fly quantization you simply need to pass a the quantization type for TGI and it takes care of the rest.
```
Comment (pcuenca) [inline docs/source/conceptual/quantization.md:12]:
```suggestion
For on-the-fly quantization you simply need to pass one of the supported quantization types and TGI takes care of the rest. 
```

File: docs/source/conceptual/quantization.md [hunk 6]
```diff
@@ -1,6 +1,40 @@
...
+
+```bash
+docker run --gpus all --shm-size 1g -p 8080:80 -v $volume:/data ghcr.io/huggingface/text-generation-inference:latest --model-id $model --quantize bitsandbytes-nf4
+```
+
+You can get more information about 8-bit quantization by reading this [blog post](https://huggingface.co/blog/hf-bitsandbytes-integration), and 4-bit quantization by reading [this blog post](https://huggingface.co/blog/4bit-transformers-bitsandbytes).
+
+Use `eetq` or `fp8` for other quantization schemes.
+
+In addition to this, TGI allows creating GPTQ quants directly by passing the model weights and a calibration dataset.
```
Comment (pcuenca) [inline docs/source/conceptual/quantization.md:37]: Maybe add new sections for the others instead?

**Review by pcuenca — COMMENTED** [review]
Very cool, added a couple of nits 🔥

**Review by julien-c — DISMISSED** [review]
nice!

**Review by drbh — APPROVED** [review]
lgtm! Thank you @Vaibhavs10 🙏🙏

**Pull request merged**